### PR TITLE
Json

### PR DESCRIPTION
**Pull request by cholcombe973**
Add json formatting support to the commands.

### CODE REVIEW

**Comment by cholcombe973** [comment]
Example output:

```
root@juju-56a1a1df-ec79-4e92-8061-6d01c40a96df-machine-6:~# ./ceph-safe-disk -q -j
{"Safe to remove an OSD": true}
./ceph-safe-disk -e -j
{"Not removable":[],"Removable":[0,1,2],"Pending":[]}

```

File: lib/diag.rs [hunk 1]
```diff
@@ -85,7 +93,7 @@ impl DiagMap {
     // OSD or not.
     // `osd_diag` holds an OSD's removability status. Using a binary heap we
     // can always know which state it has that holds the highest precedent.
-    pub fn exhaustive_diag(self) -> Status {
+    pub fn exhaustive_diag(self, json_format: bool) -> Status {
```
Comment (ChrisMacNaughton) [inline lib/diag.rs:96]: Would it be interesting to make a Format enum, with the idea that some people may prefer different types just like juju? ie: yaml, json, tabular, etc?

Comment (ChrisMacNaughton) [inline lib/diag.rs:96]: I'd also be tempted to refactor this to have the various diags return an object that we can then output in main() to reduce repetitive rendering :) maybe the struct would `impl render(Format)`?

**Comment by ChrisMacNaughton** [comment]
As we discussed a bit, I like the JSON output; however, I think we could be a little bit more forward looking towards _how_ and _where_ we actually do the outputting :smile:

**Comment by ChrisMacNaughton** [comment]
Also, I'm not a big fan of pull requests for one feature having style changes in them :)

**Comment by cholcombe973** [comment]
Yeah sorry about that.  I should've moved that over to another PR

**Comment by 0X1A** [comment]
So, this kinda bubbled up an issue I ran into when writing `exhaustive_diag` where I kind of assumed it wasn't going to be extended. A flag for output would've been suitable for this but I didn't have the foresight to include one. Over this weekend I can take a good crack at getting something like this in 😎

**Pull request closed without merging**